databallpy/utils/synchronise_tracking_and_event_data.py

```diff
@@ -194,6 +194,9 @@ def _create_sim_mat(
                 tracking_batch["ball_x"] - tracking_batch[f"{column_id_player}_x"],
                 tracking_batch["ball_y"] - tracking_batch[f"{column_id_player}_y"],
             )
+            # player indicated by the event data is not present in the tracking data
+            if player_ball_diff.isnull().all():
+                player_ball_diff = 0
         else:
```

Comment:
Have you checked when this happens? Is this due to bad data? Should the player be present in the tracking data but he is not, or is there happening something else? If so, what data providers are you using?

Reply:
@DaanGro, do you agree this is handled well? What happens if not all values are null? is the `_needleman_wunch` able to handle `NaN` values?

Reply:
Yep so this seems to be due to bad data. I am using tracab. I only found this to happen for a single recent Eredivisie game, namely `20230513FCUtrechtvsRKCWaalwijk2292010`, so please feel free to test this yourself if you have access to the data.
Essentially from frame `2014575` to frame `2031030` there are no recorded player positioning, despite the ball positioning being there and the `ball_status` being `alive`. Also the `matchtime_td` is flowing during that time making it get to `90:00+30:29` by the end of the match. 

Something is going on with this match, but I am yet to determine if it's really the raw tracab data that's problematic or how databallpy parses it into `match.tracking_data`. If the latter is to blame, I will make sure to open another issue.

Reply:
Thanks for providing some extra info, I know this match was interupted twice and have not checked yet how tracab handles that. I would assume that there is just a long time of `ball_status` being `dead` in the interuptions. However, I have not checked if this is the case. I the only other thing I can think of is that tracab forgot to assign players when the match started again, but I have not checked this yet either. I'll see if I have some time to check the data comming days, but if you have time to investigate the precise problem that would be most welcome.

Reply:
I think regarding Alex's first question, what happens when not all values are null: the _needleman_wunsch should be able to handle the NaN values. The frames where there is a NaN value will never be linked to an event as tmax is never NaN. @rdghe, could you confirm this by running synchronise_tracking_and_event_data with n_batches_per_half set to 1? 

Regarding this specific match, I think the synchronisation will not be optimal with n_batches_per_half set to the default 100. Since this assumes that the mismatch between the timing of the event in event and tracking data is not bigger than approximately 30 seconds. I don't think we have to change the default n_batches_per_half since what happened in this match is such a specific case.

On how to handle this case: setting the player_ball_diff to 0 is the same as we did for events that do not have a player associated (line 201). This "attracts" the algorithmn to these frames as it corresponds to perfect alignment between the tracking and event data. Perhaps a better way to handle this is by setting the player_ball_diff to be the mean of the other two variables of the similarity function so that the algorithmn is not necessarily attracted to the frames where player_ball_diff is 0. Let me know, what you think about this and whether I missed something.